data_structures/priority_queue.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ * @author [Tuan Ho](https://github.com/ttuanho)
+ * @brief Implementation of [Priority
+ * queue](https://en.wikipedia.org/wiki/Priority_queue) an abstract data type
+ * similar to regular queue or stack data structure in which each element
+ * additionally has a "priority" associated with it.
+ * @note improvised from queue.h
+ * @see queue.h
+ */
+
+#include <cassert>
+#include <iostream>
+
+using Priority = int;
+
+static constexpr int LOWEST_PRIORITY = -1;
+
+/** Definition of the node */
+template <class Kind, class Priority>
+struct node {
+    Kind data;
+    node<Kind, Priority> *next;
+    Priority prior;  // priority level which must be >= 0
+};
+
+/** Definition of the priority queue class */
+template <class Kind, class Priority>
+class queue {
+ public:
+    /** Show queue */
+    void display() {
+        node<Kind, Priority> *current = queueFront;
+        std::cout << "Front --> ";
+        while (current != NULL) {
```

Comment:
C++ algorithms should always use `nullptr`.
Please check other parts of the code for this.
```suggestion
        while (current != nullptr) {
```

Reply:
Yup. Just pushed.